interfaces/apparmor/spec_test.go

```diff
@@ -537,6 +537,26 @@ func (s *specSuite) TestApparmorExtraLayouts(c *C) {
 	// lines 3..9 is the traversal of the prefix for /usr/home/test
 }
 
+func (s *specSuite) TestAllowEnsureDirMounts(c *C) {
+	ensureDirSpecs := []*interfaces.EnsureDirSpec{
+		{MustExistDir: "$HOME", EnsureDir: "$HOME/.local/share"},
+		{MustExistDir: "$HOME", EnsureDir: "$HOME/dir1/dir2"},
+		{MustExistDir: "/", EnsureDir: "/dir1/dir2"},
+		{MustExistDir: "/dir1", EnsureDir: "/dir1"},
+	}
+	s.spec.AllowUserEnsureDirMounts("personal-files", ensureDirSpecs)
```

Comment:
it's a bit strange that a function with user in the name let us manipulate / ? maybe it should error or ignore on things that don't have MustExistDIr == $HOME ?

Reply:
I started with a generic function that covers both user and system use of ensure-dir. Then later tried to indicate the currently intended scope by adding ...User...

Thinking about it again, I see no reason why it cannot be a generic function. I added additional comments to show where additional expansions should be added on a need to have basis.

Reply:
As discussed: 
- avoid errors on the apparmor spec emit side
- check for errors on the mount spec side

Proposal: I added a definitive validation function for ensure-dir specs, that is used on the mount side when the interface attempts to add mount entries on connected plug. It result in an internal error if for whatever reason the interface creating the ensire-dir specs breaks the rules. If this is too much, I am happy to remove.